watertap/examples/flowsheets/case_studies/activated_sludge/tests/test_modified_asm2d_flowsheet.py

```diff
@@ -0,0 +1,171 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2023, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+"""
+Tests for modified ASM2D flowsheet example.
```

Comment:
I'm not sure if we want to start doing this consistently with our test files yet, but in IDAES, we are starting to add tests that use Andrew's Diagnostic Toolbox to ensure there are no structural or numerical issues with the system. I think it's worthwhile to begin adding such tests. Here's an example in IDAES: https://github.com/IDAES/idaes-pse/blob/main/idaes/models/unit_models/solid_liquid/tests/test_sl_separator.py

Reply:
The structural issues check can replace the existing degrees of freedom check if you like (as it includes a check for DoF).

Reply:
For that matter, the structural issue check also replaces the unit consistency check too.

Reply:
> For that matter, the structural issue check also replaces the unit consistency check too.

@andrewlee94 There is a parameter `f_SI` in the `modified_asm2d_reactions.py` packageis fixed at 0 and its domain is `NonNegativeReals`, however it is always reported as a numerical issue (variable outside the bound) with the `DiagnosticsToolbox`. I think we should exclude this situation as a numerical problem, otherwise the tests cannot pass.

Reply:
First, you can change the tolerances for the checks, and second that is expected behaviour.

Reply:
> First, you can change the tolerances for the checks, and second that is expected behaviour.

@andrewlee94 can you clarify how to change the tolerance for avoiding failure of `assert_no_numerical_warnings()` due to this case?

Reply:
The DiagnosticsToolbox has a config block with all the tolerances. You can set these via keyword arguments when instantiating the DiagnosticsToolbox, or by directly changing the config block.

Reply:
@luohezhiming I think @andrewlee94 was referring to this or maybe another config option: https://github.com/IDAES/idaes-pse/blob/main/idaes/core/util/model_diagnostics.py#L195-L203

Reply:
> @luohezhiming I think @andrewlee94 was referring to this or maybe another config option: https://github.com/IDAES/idaes-pse/blob/main/idaes/core/util/model_diagnostics.py#L195-L203

Yeah, I've already addressed this with the tolerance set as 1e-9